crates/proc-macro-api/src/process.rs

```diff
@@ -19,19 +19,45 @@ pub(crate) struct ProcMacroProcessSrv {
     _process: Process,
     stdin: ChildStdin,
     stdout: BufReader<ChildStdout>,
+    version: u32,
 }
 
 impl ProcMacroProcessSrv {
     pub(crate) fn run(
         process_path: AbsPathBuf,
-        args: impl IntoIterator<Item = impl AsRef<OsStr>>,
+        args: impl IntoIterator<Item = impl AsRef<OsStr>> + Clone,
     ) -> io::Result<ProcMacroProcessSrv> {
-        let mut process = Process::run(process_path, args)?;
-        let (stdin, stdout) = process.stdio().expect("couldn't access child stdio");
+        let create_srv = || {
+            let mut process = Process::run(process_path.clone(), args.clone())?;
+            let (stdin, stdout) = process.stdio().expect("couldn't access child stdio");
+
+            io::Result::Ok(ProcMacroProcessSrv { _process: process, stdin, stdout, version: 0 })
+        };
+        let mut srv = create_srv()?;
+        tracing::info!("sending version check");
+        match srv.version_check() {
+            Ok(v) => {
+                tracing::info!("got version {v}");
+                srv.version = v;
```

Comment:
Maybe add a check that the version is not newer than that of rust-analyzer itself?